sale_procurement_group_by_line/model/sale.py

```diff
@@ -77,23 +62,54 @@ def _action_procurement_create(self):
                     groups[l._get_procurement_group_key()] = g_id
             if not group_id:
                 group_id = groups.get(line._get_procurement_group_key())
+
             if not group_id:
                 vals = line.order_id._prepare_procurement_group_by_line(line)
-                group_id = self.env["procurement.group"].create(vals)
+                vals.update({
+                    'move_type': line.order_id.picking_policy,
+                    'sale_id': line.order_id.id,
+                    'partner_id': line.order_id.partner_shipping_id.id
+                })
+                group_id = self.env['procurement.group'].create(vals)
+            else:
+                # In case the procurement group is already created and the
+                # order was cancelled, we need to update certain values
+                # of the group.
+                updated_vals = {}
+                if group_id.partner_id != line.order_id.partner_shipping_id:
+                    updated_vals.update({'partner_id':
+                                         line.order_id.partner_shipping_id.id})
+                if group_id.move_type != line.order_id.picking_policy:
+                    updated_vals.update({'move_type':
+                                         line.order_id.picking_policy})
+                if updated_vals:
+                    group_id.write(updated_vals)
             line.procurement_group_id = group_id
 
-            vals = line._prepare_order_line_procurement(
-                group_id=line.procurement_group_id.id)
-            vals['product_qty'] = line.product_uom_qty - qty
-            new_proc = self.env["procurement.order"].create(vals)
-            new_proc.message_post_with_view(
-                'mail.message_origin_link',
-                values={'self': new_proc, 'origin': line.order_id},
-                subtype_id=self.env.ref('mail.mt_note').id)
-            new_procs += new_proc
-        new_procs.run()
-        super(SaleOrderLine, self)._action_procurement_create()
```

Comment:
Why did you delete this super?

Reply:
@mreficent Because In odoo v11 _action_procurement_create method is removed and added _action_launch_procurement_rule method [https://github.com/odoo/odoo/commit/c914df5a369eb49e4e72a698c0811200b5825f22#diff-bf6a9af0e4d7537746d73deb353f3f8aL210](url) so I have overwrite this method and added code.

Reply:
Yeah, but why you remove the super? You can do super with the _action_launch_procurement_rule method.

Reply:
@mreficent [Here](https://github.com/odoo/odoo/commit/c914df5a369eb49e4e72a698c0811200b5825f22#diff-bf6a9af0e4d7537746d73deb353f3f8aR235) super return `True` , So How can modify the code with super?.

That's way @murtuzasaleh overwrite code and modify that.

Here Super is not need. @aheficent already https://github.com/OCA/sale-workflow/pull/673#discussion_r233780204

Reply:
```python
return super(self, SaleOrderLine)._action_launch_procurement_rule()
```

Reply:
Hi @murtuzasaleh @nikul-serpentcs Did you try to call super at the end? what does the system do in that case? Are other procurement groups created or the ones you create updated? Maybe they are not. I am not sure looking at the code.

Reply:
>  Did you try to call super at the end?

@aheficent Yes, But it's method Call again.

Reply:
But does that affect the result?